server/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java

```diff
@@ -73,14 +90,14 @@ protected TransportNodesAction(
         ActionFilters actionFilters,
         Writeable.Reader<NodesRequest> request,
         Writeable.Reader<NodeRequest> nodeRequest,
-        String executor
+        Executor executor
     ) {
```

Comment:
I believe this constructor still has `ThreadPool threadPool` parameter that is now unused. Should it be removed?

Reply:
+1 but in a follow-up I think, I'd like to limit the scope of this PR as much as possible.